src/nvim/option.c

```diff
@@ -3822,8 +3865,25 @@ set_bool_option (
    * End of handling side effects for bool options.
    */
 
+  // after handling side effects, call autocommand
+
   options[opt_idx].flags |= P_WAS_SET;
 
+  if (!starting) {
+    char_u buf_old[2], buf_new[2], buf_type[7];
+    vim_snprintf((char *)buf_old, 2, "%d", old_value ? true: false);
+    vim_snprintf((char *)buf_new, 2, "%d", value ? true: false);
```

Comment:
By the way, `true` and `false` are have single byte `_Bool` type. `%d` expects 16, 32 or 64 (system-dependent) `int`. I think you need a cast just in case.